azure-keyvault/tests/test_keys.py

```diff
@@ -0,0 +1,242 @@
+        key_bundle = client.update_key(
+            key.name, key.version,
+            expires=expires,
+            tags=tags)
+        self.assertEqual(tags, key_bundle.tags)
+        self.assertEqual(key.id, key_bundle.id)
+        self.assertNotEqual(key.updated, key_bundle.updated)
+        return key_bundle
+
+    def _validate_key_list(self, keys, expected):
+        for key in keys:
+            if key.name in expected.keys():
+                del expected[key.name]
+        self.assertEqual(len(expected), 0)
+
+    @ResourceGroupPreparer()
+    @KeyVaultPreparer(enable_soft_delete=True)
+    def test_key_crud_operations(self, vault_client, **kwargs):
+
+        self.assertIsNotNone(vault_client)
+        client = vault_client.keys
+        key_name = 'crud-test-key'
+
+        # create key
+        created_key = client.create_key(key_name, 'RSA')
+        self.assertIsNotNone(created_key.version)
+        self._validate_rsa_key_bundle(created_key, vault_client.vault_url, key_name, 'RSA')
+        # assert it creates a version on creation of the key
+        self.assertTrue(len(list(client.list_key_versions(key_name))) == 1)
+
+        # create key with optional arguments
+        key_size = 2048
+        key_ops = [
+            "encrypt",
+            "decrypt",
+            "sign",
+            "verify",
+            "wrapKey",
+            "unwrapKey"
+        ]
+        tags = {"purpose": "unit test", "test name ": "CreateGetDeleteKeyTest"}
+        created_key = client.create_key(key_name, 'RSA', size=key_size, key_ops=key_ops, tags=tags)
+        self.assertTrue(created_key.version and created_key.tags, 'Missing the optional key attributes.')
+        self.assertEqual(tags, created_key.tags)
+        self._validate_rsa_key_bundle(created_key, vault_client.vault_url, key_name, 'RSA')
+
+        # get the created key with version
+        key = client.get_key(key_name, created_key.version)
+        self.assertEqual(key.version, created_key.version)
+        self._assert_key_attributes_equal(created_key, key)
+
+        # get key without version
+        self._assert_key_attributes_equal(created_key, client.get_key(created_key.name, ''))
+
+        # update key with version
+        if self.is_live:
+            # wait to ensure the key's update time won't equal its creation time
+            time.sleep(1)
+        updated = self._update_key(client, created_key)
+
+        # delete the new key
+        deleted_key = client.delete_key(key_name)
+        self.assertIsNotNone(deleted_key)
+        self.assertEqual(created_key.key_material, deleted_key.key_material)
+        self.assertEqual(deleted_key.id, created_key.id)
+        self.assertTrue(deleted_key.recovery_id and deleted_key.deleted_date and deleted_key.scheduled_purge_date,
+                        'Missing required deleted key attributes.')
+
+        if self.is_live:
+            # wait to ensure the key has been deleted
+            time.sleep(20)
```

Comment:
Should we have this method placed somewhere more globally to be accessed in both test_keys and test_example_keys, so that we don't have to have it in both the places?

Reply:
As discussed, Would be doing this once we have the latest changes of azure-core.